src/marqo/core/semi_structured_vespa_index/semi_structured_vespa_index.py

```diff
@@ -156,6 +158,9 @@ def generate_equality_filter_string(node: search_filter.EqualityTerm) -> str:
             return final_filter_string
 
         def generate_range_filter_string(node: search_filter.RangeTerm) -> str:
+            # Escape special characters in field name
+            node.field = self.escape(node.field)
```

Comment:
why do we need to escape the special characters in field names when field name should not contain any special character?

Reply:
For structured indexes yes, we don't need it technically, but for queries on an unstructured (semi-structured) index, it's still possible to filter on a field that doesn't technically exist in the index. So query will still be sent to vespa

Reply:
So we choose to send the query and return 0 hits, instead of erroring out here, right?

Reply:
Yep exactly, it's the same behavior in unstructured when you try to filter on a field that doesn't exist